File: ios/PIP/PipModule.m
#import <Foundation/Foundation.h>

#if __has_include(<React/RCTBridgeModule.h>)
#import <React/RCTBridgeModule.h>
#elif __has_include("RCTBridgeModule.h")
#import "RCTBridgeModule.h"
#else
#import "React/RCTBridgeModule.h"
#endif

@interface RCT_EXTERN_MODULE(PipModule, NSObject)

RCT_EXTERN_METHOD(setEnabled:(BOOL)enabled)
RCT_EXTERN_METHOD(enterPipMode)

@end

File: ios/PIP/PipModule.swift
import Foundation

@objc(PipModule)
class PipModule: NSObject {
  
  @objc
  static func requiresMainQueueSetup() -> Bool {
    return true
  }
  
  @objc
  func setEnabled(_ enabled: Bool) {
    // iOS handles PiP state mostly via the player itself
    // We provide this method for parity with Android
    print("PiP Enabled set to: \(enabled)")
  }
  
  @objc
  func enterPipMode() {
    // on iOS, PiP is usually triggered by backgrounding the app or via the player UI
    // Manual triggering from JS is possible but usually done through the Video component
    print("enterPipMode called on iOS")
  }
}
